addons/test_mail/tests/test_mail_composer.py

```diff
@@ -453,6 +453,43 @@ def test_mail_composer_rights_attachments(self):
             sorted(self.test_record.message_ids[0].attachment_ids.mapped('name')),
             sorted(template_1_attachment_name))
 
+    @users('employee')
+    def test_mail_composer_template_switching(self):
+        """ Checks switching templates and effect it has on `email_from` and `subject` field (as well as subject).
+        Main goal of this test is to ensure that current user is used as the sender,
+        when the template is changed from template with sender to the one without it"""
+        # Setup
+        template_with_sender_and_subject = self.template.copy({'email_from': 'not_current_user@test.com','subject': 'First Subject'})
+        template_without_sender = template_with_sender_and_subject.copy({'email_from': ''})
+        template_without_subject_nor_body = template_without_sender.copy({'subject': '', 'body_html': ''})
+        template_with_other_subject_and_body = template_without_sender.copy({'subject': 'Other Subject', 'body_html': 'Other Body'})
+        form = Form(self.env['mail.compose.message'])
+
+        for composition_mode in ['comment','mass_mail']:
+            with self.subTest(msg='subtest per composition_mode', composition_mode=composition_mode):
+                # Step 0: use template with sender and with subject
+                form.composition_mode = form.composition_mode
+                form.template_id = template_with_sender_and_subject
+                self.assertEqual(form.email_from, template_with_sender_and_subject.email_from,
+                    'email_from not set correctly to form this test')
+                self.assertEqual(form.subject, template_with_sender_and_subject.subject, 'subject not set correctly to form this test')
+
+                # Step 1: switch to template without sender
+                form.template_id = template_without_sender
+                self.assertEqual(form.email_from, self.env.user.email_formatted, 'email_from not updated to current user')
+
+                # Step 2: switch to template without subject nor body
+                form.template_id = template_without_subject_nor_body
+                self.assertFalse(form.subject, 'switching to template without a subject shall unset subject')
+                self.assertFalse(form.body, 'switching to template without a body shall unset body')
+
+                # Step 3: switch to template with a different subject and body
+                form.template_id = template_with_other_subject_and_body
+                self.assertEqual(form.subject, template_with_other_subject_and_body.subject,
+                    'switching to template with a subject shall update it')
+                self.assertEqual(form.body, template_with_other_subject_and_body.body_html,
+                    'switching to template with some body shall update it')
```

Comment:
Maybe you can keep those and make it explicit "roll back" is not supported for body/subject.

Reply:
I just narrowed down the fix to handle only the email_from and reply_to fields. However, I'm unsure how to tweak the test to reflect the behavior of body and subject, as it would need context insertion. Yet, relying on frontend context might lead to irrelevant tests if the frontend changes its behavior later.

Reply:
I meant you could just check that setting a template set the subject, but setting another template without subject keeps the same subject. Just as a documentation step.

But not testing it is fine actually, it doesn't really matter what we do with it.